### PR TITLE
Common Lisp Client

### PR DESCRIPTION
**Pull request by Dotrar**
Continuing from the work greg's branch, I've implemented a fairly rudimentary but reliable (i hope) common lisp client. 

### Edit: 10th Jan
STDOUT now uses comment prefix to show up in the log. picture in [this comment](https://github.com/Olical/conjure/pull/296#issuecomment-1008428451) is what it looks like.

### Old picture, not quite correct

I've tried to keep it atomic, compiled after every commit, etc.

I'm out on holidays at the moment, and I have to walk 1.5 kms to get reception, so I probably won't reply until monday, but feel free to have a look and poke as many holes in it as possible and I can get them fixed up next week.

What defines a "client" that is ready to use? I've just gone for eval / stdout / errors here.That should be enough to try and build upon in the future?

### CODE REVIEW

**Comment by Dotrar** [comment]
this should complete #97 in my estimation. 

some questions:
- Do you want some docs to go with this? I was thinking of writing out "what does a client need" sort of thing, and basic api info, but not sure what should be doc'd. (i just copied things) - this might be better in another PR.
- How is stdout meant to be shown in the conjure log? I just used a string representation, which seems good enough.  (i didn't check escape values, whoops)
- Is there any plans for stdin ? I was thinking something like the vim input command could work for simple cases, but could be hard to do

**Comment by rafaeldelboni** [comment]
Thanks a lot for this! 😀

File: fnl/conjure/config.fnl [hunk 1]
```diff
@@ -48,18 +48,19 @@
     nil))
 
 (merge
-  {:debug false
+  {:debug true
```
Comment (rafaeldelboni) [inline fnl/conjure/config.fnl:51]: I think you need to toogle this back to false right?

File: fnl/conjure/remote/swank.fnl [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                      (opts.on-failure err)
+
+                      (do
+                        (conn.sock:read_start (client.schedule-wrap handle-message))
+                        (opts.on-success)))))})))
+
+  ; (send conn (or opts.name "Conjure"))
+  conn)
+
+; (def c (connect {:host "127.0.0.1"
```
Comment (rafaeldelboni) [inline fnl/conjure/remote/swank.fnl:58]: Maybe cleanup this comments?

File: fnl/conjure/config.fnl (hunk 1 above)
Comment (Dotrar) [reply fnl/conjure/config.fnl]: Yep! slipped by from greg's work. I've fixed this in [07159d7856ac7f6d8eb3103d331d5d1e5d7b734c]

File: fnl/conjure/remote/swank.fnl (hunk 2 above)
Comment (Dotrar) [reply fnl/conjure/remote/swank.fnl]: also fixed in [07159d7856ac7f6d8eb3103d331d5d1e5d7b734c]

**Comment by Dotrar** [comment]
Added doc word and eval-file. 
image

**Comment by Olical** [comment]
I've still never used common lisp in any capacity so I'm taking your word for it when it comes to UX and behaviour right now but this looks fantastic. Excellent work! I'll have a look through the code from a Fennel / maintainer point of view but I might end up "LGTM"-ing some of the more nuanced CL parts since it's not my area yet.

I'll try to at least get a Hello World eval working though! I'll resolve the conflicts too.

**Comment by Olical** [comment]
And thank you @rafaeldelboni for swooping in with early reviews and 👀 !

**Comment by Olical** [comment]
At the surface level you've nailed the general file structure and my rough architecture. I'm going to merge-aroo and try it out locally, if things look okay let's get people trying it out on the develop branch for a couple of days then time for a release. There's a lot on the develop branch I want to get out right now.

**Comment by Olical** [comment]
Took me a while to get a swank REPL up but it works for me! Nice! All the base features are there, I'll just knock up a basic help .txt but if someone wants to create the quickstart wiki page that'd be amazing. I think someone who actually knows some good CL install and setup tips would be best placed to do that though.

**Comment by Olical** [comment]
🚀  now let's let the dust settle slightly on develop and then I'll get this big ol' batch of commits tagged as stable soon.

**Pull request merged**